src/components/ColorPickerContainer.ts

```diff
@@ -0,0 +1,286 @@
+            if (onChangeEvent === "callMicroflow" && onChangeMicroflow) {
+                window.mx.ui.action(onChangeMicroflow, {
+                    error: error => window.mx.ui.error(`Error while executing microflow ${onChangeMicroflow}: ${error.message}`), // tslint:disable-line max-line-length
+                    params: {
+                        applyto: "selection",
+                        guids: [ mxObject.getGuid() ]
+                    },
+                    origin: mxform
+                });
+            } else if (onChangeEvent === "showPage" && onChangePage) {
+                window.mx.ui.openForm(onChangePage, {
+                    context,
+                    error: error => window.mx.ui.error(`Error while opening page ${onChangePage}: ${error.message}`),
+                    location: this.props.openPageLocation
+                });
+            } else if (onChangeEvent === "callNanoflow" && onChangeNanoflow.nanoflow) {
+                window.mx.data.callNanoflow({
+                    context,
+                    error: error => window.mx.ui.error(`Error while executing the onchange nanoflow: ${error.message}`),
+                    nanoflow: onChangeNanoflow,
+                    origin: mxform
+                });
+            }
+        }
+    }
+
+    public static validateProps(props: ColorPickerContainerProps): string {
+        const message: string[] = [];
+        if (props.onChangeEvent === "callMicroflow" && !props.onChangeMicroflow) {
+            message.push("On change event is set to 'Call a microflow' but no microflow is selected");
+        } else if (props.onChangeEvent === "showPage" && !props.onChangePage) {
+            message.push("On change event is set to 'Show a page' but no page is selected");
+        } else if (props.onChangeEvent === "callNanoflow" && (JSON.stringify(props.onChangeNanoflow) === JSON.stringify({}))) {
+            message.push("On change event is set to 'Call a nanoflow' but no nanoflow is selected");
+        }
+        if (props.label.trim() && props.labelWidth > 11) {
+            message.push("Label width (weight) should not be greater than 11");
+        }
+
+        return message.length ? `Configuration error in widget ${props.friendlyId}: ${message.join(", ")}` : "";
+    }
+
+    public static parseStyle(style = ""): {[key: string]: string} {
+        try {
+            return style.split(";").reduce<{[key: string]: string}>((styleObject, line) => {
+                const pair = line.split(":");
+                if (pair.length === 2) {
+                    const name = pair[0].trim().replace(/(-.)/g, match => match[1].toUpperCase());
+                    styleObject[name] = pair[1].trim();
+                }
+                return styleObject;
+            }, {});
+        } catch (error) {
+            // tslint:disable-next-line no-console
+            console.log("Failed to parse style", style, error);
```

Comment:
Rethink with the comments above...
lets discuss

Reply:
Yeah we should discuss this